src/data/init/init-abilities.ts

```diff
@@ -1622,29 +1621,29 @@ export function initAbilities() {
       .attr(PostAttackApplyStatusEffectAbAttr, false, 30, StatusEffect.TOXIC)
       .edgeCase(), // Does not inflict poison if user gets inflicted with target's Mummy
     new Ability(AbilityId.EMBODY_ASPECT_TEAL, 9)
-      .attr(PostBattleInitStatStageChangeAbAttr, [Stat.SPD], 1, true)
+      .attr(PostTeraFormChangeStatChangeAbAttr, [Stat.SPD], 1)
+      .attr(PostSummonStatStageChangeAbAttr, [Stat.SPD], 1)
```

Comment:
```suggestion
      .attr(PostSummonStatStageChangeAbAttr, [Stat.SPD], 1, true)
```
(Sets `selfTarget` to `true`)